src/learn/gensfen.cpp

```diff
@@ -248,7 +262,15 @@ namespace Learner
             // When parallelizing, Threads (since this is a vector<Thread*>,
             // Do the same for up to Threads[0]...Threads[thread_num-1].
             auto& pos = th.rootPos;
-            pos.set(StartFEN, false, &si, &th);
+            if (opening_book != nullptr)
```

Comment:
[nit pick] We can write `if (opening_book) {` here because std::unique_ptr<T> implements `operator bool()`.